### PR TITLE
update UIImage, UITextField extensions

### PR DESCRIPTION
**Pull request by xingfuqiu**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 3.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [X] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.

### CODE REVIEW

File: Sources/Extensions/UIKit/UITextFieldExtensions.swift [hunk 1]
```diff
@@ -79,14 +79,27 @@ public extension UITextField {
...
+    ///
+    /// - Parameter padding: amount of padding to apply to the left of the textfield rect.
+    public func addPaddingLeft(_ padding: CGFloat) {
     let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: padding, height: frame.height))
     leftView = paddingView
     leftViewMode = .always
-  }
+    }
+    
+    /// Add padding to the left of the textfield rect.
```
Comment (SD10) [inline Sources/Extensions/UIKit/UITextFieldExtensions.swift]: @xingfuqiu All extensions need to be marked with SwifterSwift in the description

**Comment by SD10** [comment]
@xingfuqiu Are you willing to write Unit Tests for this new extension?

File: Sources/Extensions/UIKit/UITextFieldExtensions.swift [hunk 2]
```diff
@@ -79,14 +79,27 @@ public extension UITextField {
...
+    ///
+    /// - Parameters:
+    ///   - image: left image
+    ///   - padding: image.size + padding, default: 3
+    public func addPaddingLeftIcon(_ image: UIImage, padding: CGFloat = 3) {
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/UIKit/UITextFieldExtensions.swift]: @xingfuqiu Any reason for a default value here, and for the value 3?

File: Sources/Extensions/UIKit/UITextFieldExtensions.swift (hunk 1 above)
Comment (xingfuqiu) [inline Sources/Extensions/UIKit/UITextFieldExtensions.swift]: ok

File: Sources/Extensions/UIKit/UITextFieldExtensions.swift (hunk 2 above)
Comment (xingfuqiu) [inline Sources/Extensions/UIKit/UITextFieldExtensions.swift]: fix it

File: Sources/Extensions/UIKit/UIImageExtensions.swift [hunk 3]
```diff
@@ -137,26 +137,26 @@ public extension UIImage {
...
-			return
-		}
-		self.init(cgImage: aCgImage)
-	}
+    /// SwifterSwift: Create UIImage from color and size.
+    ///
+    /// - Parameters:
+    ///   - color: image fill color.
+    ///   - size: image size. default: 1x1
+    public convenience init(color: UIColor, size: CGSize = CGSize(width: 1, height: 1)) {
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/UIKit/UIImageExtensions.swift]: Hey @xingfuqiu , I don't know if  a default value is a god fit for this init, do you have any reason for that?

Comment (xingfuqiu) [inline Sources/Extensions/UIKit/UIImageExtensions.swift]: i think it will be easy to use

Comment (LucianoPAlmeida) [inline Sources/Extensions/UIKit/UIImageExtensions.swift]: I think a default value for size in this case, seems like a random value, and i don't see a use case where (1,1) as default will be used. Can you change that?

**Comment by SD10** [comment]
Continued in #203

**Pull request closed without merging**